src/collective/taxonomy/jsonimpl.py

```diff
@@ -47,6 +52,7 @@ def generate_json(self, root):
 
     def get_data(self):
         """Get json data."""
+        alsoProvides(self.request, IDisableCSRFProtection)
```

Comment:
Why is this necessary here? No ZODB object seems to be modified in this code.

Reply:
I have do dig here further, but if you have a clean Plone-5.1 installation, create a taxonomy and want to edit data for the first time, you get a CSRF warning from plone.protect ...

Reply:
Maybe it happens in the `_fixup` method.

You could argue that there shouldn't be any fixing upping on a `GET` request.

Reply:
thanks for the pointer ... the `_fixup` was indeed the problem ... I've refactored this to get rid of CSRF warnings ...

Reply:
The dilemma is this:

There's an old version of the data format and there is an upgrade step which migrates this to the new format by calling `_fixup()`.

But what if people try to use their site without – yet – having gone through the upgrade step.

I think that was the reasoning behind these fixing upper calls. And there used to be no upgrade step!

But now, perhaps the right move is simply to throw and error and log a warning "Data not upgraded" – ?